ygen/codegen.go

```diff
@@ -885,11 +896,25 @@ type mappedYANGDefinitions struct {
 // It returns a mappedYANGDefinitions struct populated with the directory, enum
 // entries in the input schemas as well as the calculated schema tree.
 func mappedDefinitions(yangFiles, includePaths []string, cfg *GeneratorConfig) (*mappedYANGDefinitions, util.Errors) {
-	modules, errs := processModules(yangFiles, includePaths, cfg.ParseOptions.YANGParseOptions)
+	moduleEntries, modules, errs := processModules(yangFiles, includePaths, cfg.ParseOptions.YANGParseOptions)
 	if errs != nil {
 		return nil, errs
 	}
 
+	// Exclude any modules that do not appear in the yangFiles list.
+	if cfg.ExcludeSearchPathModules {
+		inFiles := map[string]bool{}
```

Comment:
If we decide to make this the standard behavior then I agree. But otherwise, you have to build the included map with all module when not ExcludeSearchPathModules is false and modify findMappableEntities to take included modules as well. 
I don't feel very strongly about this, so whatever you prefer.

Reply:
No I think your solution is good.